lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -387,6 +386,17 @@ def _on_opensearch_data_storage_detaching(self, _: StorageDetachingEvent):  # no
         # acquire lock to ensure only 1 unit removed at a time
         self.ops_lock.acquire()
 
+        def _get_cluster_color():
```

Comment:
Let's take this part with regards to scale-down out of this PR and mere our efforts on this (the issue we talked about last week)